### DISCUSSION TITLE
Hardware resource requirement for local use

### DISCUSSION THREAD

**Discussion by impredicative** [microsoft/graphrag#325] [Q&A] [answered]
I am not exactly sure how graphrag works or what it does behind the scenes.

When running locally, how much hardware does it require for:

1. Graph generation
2. Answering questions using the generated graph

Also, for each these, if an LLM is used, how do I get it to use GPT4o?

**Comment by AlonsoGuevara** [comment, marked as answer]
Hi @impredicative 

As far as how much hardware is required, processing depends on how big your dataset is, with the proper set up you should be able to run locally without major problems.
 As per configuration, I would strongly recommend to run the getting started example so you get familiar with how to set up your working environment: https://microsoft.github.io/graphrag/posts/get_started/

One of the key points of the getting started guide is that you'll run the `--init` command, it will basically bootstrap a configuration file for your runs, which is by default set up to run on OpenAI, so you should be able to use GPT4o after providing the API Key

**Comment by impredicative** [comment]
For each question, I want to myself create a dynamic dataset first, then the graphrag over it, then use it to answer a question.
